lighthouse-core/gather/computed/computed-artifact.js

```diff
@@ -59,13 +60,17 @@ class ComputedArtifact {
       return computed;
     }
 
+    const status = {msg: `Computing artifact: ${this.name}`, id: `lh:computed:${this.name}`};
+    log.time(status, 'verbose');
     // Need to cast since `this.compute_(...)` returns the concrete return type
     // of the base class's compute_, not the called derived class's.
     const artifactPromise = /** @type {ReturnType<this['compute_']>} */ (
       this.compute_(requiredArtifacts, this._allComputedArtifacts));
     this._cache.set(requiredArtifacts, artifactPromise);
 
-    return artifactPromise;
+    const artifactValue = await artifactPromise;
```

Comment:
the downside of this is that it linearizes all computed artifacts requests. Granted, I can't really think of any off-thread ones we have now, but with workers coming to node we might not want to box ourselves in.

Maybe we want to do it like
```
artifactPromise.then(_ => log.timeEnd(status));
return artifactPromise;
```
? I think that might work

Reply:
> but with workers coming to node we might not want to box ourselves in.

Do we have any concrete plans to move our work to workers? :)

Reply:
> the downside of this is that it linearizes all computed artifacts requests

oh, no, I fell for it. That's exactly equivalent. That's the whole point of forcing async on the function when using await, letting the caller decide on the asynchrony :)